Serilog.Sinks.Network/Sinks/TCP/TCPSocketWriter.cs

```diff
@@ -240,10 +240,20 @@ public async Task<Stream> ConnectAsync(Func<Uri, Task<Stream>> connect, Uri host
             {
                 try
                 {
-                    Log.Debug("Attempting to connect to TCP endpoint {host} after delay of {delay} seconds", host, delay);
+                    Log.Debug("Attempting to connect to TCP endpoint {host} after delay of {delay} seconds", host,
+                        delay);
                     return await connect(host);
                 }
-                catch (SocketException) { }
+                catch (SocketException)
+                {
+                }
+                catch (IOException e)
+                {
+                    if (e.InnerException is SocketException)

```

Comment:
should this be the reverse to maintain behaviour of swallowing socket exceptions?

Reply:
Yeah, this was missing a `not`. A bad port from my end.